### DISCUSSION TITLE
Add a `redirectTo` prop

### DISCUSSION THREAD

**Discussion by Jeysef** [47ng/nuqs#567] [Ideas] [answered]
I am using:
- Next.js: 14.2.3
- app router

I wish:
An optional `redirectTo` prop to pass to nuqs that would change the url and add the query params at the same time.

Use case:
I am using search in `layout.ts`. When someone types a name, I want nuqs to redirect him to another page and also add the query params. Because I am using the search in layout, the search is still visible after redirecting.

**Comment by franky47** [comment, marked as answer]
Have you looked into the [serializer helper](https://nuqs.47ng.com/docs/utilities#serializer-helper) to apply search params to a given path?

**Reply by Jeysef** [reply]
Yes, that helped. thanks
